src/indexes/text/fuzzy.h

```diff
@@ -1,43 +1,202 @@
+// /*
+//  * Copyright (c) 2025, valkey-search contributors
+//  * All rights reserved.
+//  * SPDX-License-Identifier: BSD 3-Clause
+//  *
+//  */
+
+// #ifndef _VALKEY_SEARCH_INDEXES_TEXT_FUZZY_H_
+// #define _VALKEY_SEARCH_INDEXES_TEXT_FUZZY_H_
+
+// namespace valkey_search::indexes::text {
+
+// //
+// // Fuzzy iterator that returns keys
+// //
+// // two stage iterator. Does FuzzyWordIterator then maps that into keys
+// //
+// struct FuzzyKeyIterator : public indexes::EntriesFetcherIteratorBase {
+//   FuzzyIterator(indexes::Text& index, absl::string_view word, size_t distance);
+//   virtual bool Done() const override;
+//   virtual void Next() = override;
+//   virtual const Key& operator*() const override;
+//   virtual std::unique_ptr<WordIterator> clone() const override;
+// };
+
+// //
+// // Takes a root word and iterates over the words that are a fuzzy match
+// //
+// struct FuzzyWordIterator : public WordIterator {
+//   FuzzyWordIterator(indexes::Text& index, absl::string_view word,
+//                     size_t distance);
+//   virtual bool Done() const override;
+//   virtual void Next() override;
+//   virtual absl::string_view GetWord() const override;
+//   virtual std::unique_ptr<WordIterator> Clone() const override;
+
+//   absl::string_view operator*() const { return GetWord(); }
+// };
+
+// };
+
+// }  // namespace valkey_search::indexes::text
+// #endif
+
 /*
  * Copyright (c) 2025, valkey-search contributors
  * All rights reserved.
  * SPDX-License-Identifier: BSD 3-Clause
- *
  */
 
 #ifndef _VALKEY_SEARCH_INDEXES_TEXT_FUZZY_H_
 #define _VALKEY_SEARCH_INDEXES_TEXT_FUZZY_H_
 
+#include <algorithm>
+#include <cctype>
+#include <string>
+#include <vector>
+#include "radix_tree.h"
+#include "absl/strings/string_view.h"
+
 namespace valkey_search::indexes::text {
 
-//
-// Fuzzy iterator that returns keys
-//
-// two stage iterator. Does FuzzyWordIterator then maps that into keys
-//
-struct FuzzyKeyIterator : public public indexes::EntriesFetcherIteratorBase {
-  FuzzyIterator(indexes::Text& index, absl::string_view word, size_t distance);
-  virtual bool Done() const override;
-  virtual void Next() = override;
-  virtual const Key& operator*() const override;
-  virtual std::unique_ptr<WordIterator> clone() const override;
-};
+// Fuzzy search using Damerau-Levenshtein distance on RadixTree
+template <typename Target>
+struct FuzzySearch {
+  // Search for words within max_distance edits of pattern
+  // Returns vector of (word, target) pairs
+  static std::vector<std::pair<std::string, Target>> Search(
```

Comment:
Looking at the existing pattern for other search types, this fuzzy search code would actually go into `BuildTextIterator()`defined on the fuzzy predicate. It would return a term iterator that holds a key iterator for every target we matched.